ili2pg/test/java/ch/ehi/ili2pg/Enum23Test.java

```diff
@@ -183,13 +183,14 @@ public void importIliWithTxtCol() throws Exception
             stmt=jdbcConnection.createStatement();          
             stmt.execute("DROP SCHEMA IF EXISTS "+DBSCHEMA+" CASCADE");
             {
-                File data=new File("test/data/Enum23/Enum23.ili");
+                File data=new File("test/data/Enum23/Enum23a.ili");
```

Comment:
besser im bestehenden Enum23.ili je einen **neuen** Aufzählwert (beim bestehenden Domain Enum1 und beim bestehenden Attribut attr2) inkl. @dispName ergänzen.

Reply:
Sälu Claude, 

bei Domain ist klar aber was meinst du mit "bestehenden Attribut attr2)"? es ist schon von Domain Enum1.

Enum23.ili angepasst:
``` 
DOMAIN
    Enum1 = (Test1, Test2_ele, Test3 (ele1, ele_2),
                !!@ili2db.dispName=testelevier
                 Test4_ele));
    
  TOPIC TestA =
    
    CLASS ClassA1 =
    	attr1 : TEXT*60;
    	attr2 : Enum1;
    	attr3 : (Test1, Test2_ele, Test3 (ele1, ele_2));
	attr4 : BOOLEAN;
    END ClassA1;
    
  END TestA;
```

merci und sorry wenn ich habe falsch verstanden.
Ruggero

Reply:
Soory, ich meinte attr3:

    DOMAIN
      Enum1 = (Test1, Test2_ele, Test3 (ele1, ele_2),
                !!@ili2db.dispName=Enum1_elevier
                 Test4_ele);
    
    TOPIC TestA =
    
     CLASS ClassA1 =
    	attr1 : TEXT*60;
    	attr2 : Enum1;
    	attr3 : (Test1, Test2_ele, Test3 (ele1, ele_2),
                !!@ili2db.dispName=Attr3_elevier
                 Test4_ele);
                   );
        attr4 : BOOLEAN;
     END ClassA1;
    
    END TestA;

Reply:
Hallo Claude,

ich habe Enum23.ili entsprechend angepasst :
```
  DOMAIN
    Enum1 = (Test1, Test2_ele, Test3 (ele1, ele_2),
            !!@ili2db.dispName=testelevier
            Test4_ele);
    
  TOPIC TestA =
    
    CLASS ClassA1 =
    	attr1 : TEXT*60;
    	attr2 : Enum1;
    	attr3 : (Test1, Test2_ele, Test3 (ele1, ele_2),
    	    !!@ili2db.dispName=Attr3_elevier
             Test4_ele);
	attr4 : BOOLEAN;
    END ClassA1;
    
  END TestA;
```
und ein neues Objekt in Enum23a.xtf hinzugefügt:
```
            <Enum23.TestA.ClassA1 TID="3">
                <attr2>Test4_ele</attr2>
                <attr3>Test4_ele</attr3>
            </Enum23.TestA.ClassA1>
```

Leider die test ch.ehi.ili2pg.Enum23Test#importXtfWithTxtCol (...import failed), ist jetzt Rot weil:
```
Error: line 21: Enum23.TestA.ClassA1: tid 3: value Test4_ele is not a member of the enumeration in attribute attr2
Error: line 21: Enum23.TestA.ClassA1: tid 3: value Test4_ele is not a member of the enumeration in attribute attr3
```

Aber der ili Modell ist korrekt importiert in die Tabellen **enum1**: 
```
itfcode,ilicode,seq,inactive,dispname,description
0,Test1,,false,Test1,
1,Test2_ele,,false,Test2 ele,
2,Test3.ele1,,false,Test3.ele1,
3,Test3.ele_2,,false,Test3.ele 2,
4,Test4_ele,,false,testelevier,
```
und **classa1_attr3**
```
itfcode,ilicode,seq,inactive,dispname,description
0,Test1,,false,Test1,
1,Test2_ele,,false,Test2 ele,
2,Test3.ele1,,false,Test3.ele1,
3,Test3.ele_2,,false,Test3.ele 2,
4,Test4_ele,,false,Attr3_elevier,
```
und der Test ch.ehi.ili2pg.Enum23Test#importIliWithTxtCol ist grün.

Ich verstehe nicht warum...sorry.
Hast du eine Idee?

https://github.com/claeis/ili2db/pull/415/commits/06da4fe1b514d3b67ae97fde49e8a37f236a8646

merci tusig